python_modules/dagster/dagster/core/execution/plan/state.py

```diff
@@ -145,9 +153,10 @@ def _create_parent_state(target_run):
                 target_run.run_id, of_type=DagsterEventType.STEP_OUTPUT
             ):
                 output_data = output_record.get_dagster_event().step_output_data
-                dynamic_outputs[output_record.step_key][output_data.output_name].append(
-                    output_data.mapping_key
-                )
+                if output_data.mapping_key:
+                    dynamic_outputs[output_record.step_key][output_data.output_name].append(
+                        output_data.mapping_key
+                    )
```

Comment:
this missing if was causing us to include add `{'step_key': {'output': [None], ...}}` for the entire dag